pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -1082,6 +1083,46 @@ func TestClusterDeploymentReconcile(t *testing.T) {
 				}
 			},
 		},
+		{
+			name: "setSyncSetFailedCondition should be present",
+			existing: []runtime.Object{
+				testInstalledClusterDeployment(time.Now()),
+				createSyncSetInstanceObj(hivev1.ApplyFailureSyncCondition),
+			},
+			validate: func(c client.Client, t *testing.T) {
+				cd := getCD(c)
+				assert.NotNil(t, cd, "cluster depployment can not be nil")
```

Comment:
Should check the return from the assert and only use `cd` if the return is `true`.